Add actions for getting 1 or 2 sockets

chris-martin opened the pull request:
Requiring exactly one or two sockets seems like a really common use case, so I thought we could support it explicitly, with good error messages for when an unexpected number of sockets is present.

I also added a re-export of the Socket type here for convenience.

ddfisher commented:
Thanks for the PR!  I'll try to find some time this week to take a more detailed look at it, but here are some initial thoughts:

- Overall, convenience methods for getting one or two activated sockets seem like a nice thing to have, so I'd be glad to merge some version of this PR.
- From a quick look, the implementation looks good.
- I'm not in love with the names for the new methods.  I like that they're quite specific, but they also feel a bit clunky/verbose.  I don't have any concrete alternatives, though.
- I'm not sure I like re-exporting `Socket` -- it doesn't feel idiomatic to me.  Are there other libraries that do similar things?

chris-martin commented:
I considered `getActivatedSockets1` and `getActivatedSockets2`, would that sound better?

Some context for why I wanted the `Socket` re-export: I was just passing the `Socket` straight into a function from another library, I ended up having to add a `network` dependency to my project just for the sole purpose of importing `Socket` so I could write an expression type-signature, which seemed sort of ridiculous. I don't really have a strong opinion, and there doesn't seem to be any popular consensus on the subject, so I'm happy to take it out if you don't want it.

chris-martin committed:
Add actions for getting 1 or 2 sockets

chris-martin commented:
Updated:

- Removed the `Socket` re-export
- Renamed the functions to `getActivatedSockets1` and `getActivatedSockets2`
- Added example usage docs for all three functions
